resources/page/site.go

```diff
@@ -147,6 +152,19 @@ func (s Sites) First() Site {
 	return s[0]
 }
 
+func (s Sites) GetSite(lang string) (Site, error) {
+	slang, err := cast.ToStringE(lang)
+	if err != nil {
+		return nil, err
+	}
+	for _, site := range s {
+		if site.Language().Lang == slang {
+			return site, nil
+		}
+	}
+	return nil, fmt.Errorf("unable to find site with lang %q", lang)
```

Comment:
I think it matches _most_ of the similar situations to return nil in this case. That way you can try to translate arbitrary languages by wrapping it in `with`.

Reply:
Thanks for the review. It looks like you have recently implemented something similar to site.GetSites while working on the pages from data stuff, so I was going to hold off until that was finalized. It looked like it I might get hugo.GetSite for free. Let me know if that's not accurate.

Reply:
No, I haven't touched anything similar in that branch. What I was concerned about in that setting was to make sure that when you do `i18n` and similar, you would get the _current_ site/language's translation, the same as you do today in `single.html` and friends.

Reply:
That said #12440 is shaping up nicely. I need to take it the final mile (resources support is missing, for one), but I have tested it out on a "real project" and it's very nice to work with (especially after I figured out how to map the `gotmpl` extension to `html` in VSCode). Having it in the templates gives the unexpected benefit of partial rebuilds "for free", so edits to assets included via `resources.Get` etc. just works. I also think this finally can make Hugo's OpenAPI integration generally useful.